cosmos/providers/databricks/notebook.py

```diff
@@ -9,9 +9,43 @@
 from airflow.providers.databricks.hooks.databricks import DatabricksHook
 from airflow.utils.context import Context
 from databricks_cli.runs.api import RunsApi
+from databricks_cli.sdk import JobsService
 from databricks_cli.sdk.api_client import ApiClient
 
 
+def _repair_task_retry_callback(context: Context) -> None:
+    """
+    This function allows the Airflow retry function to create a repair job for Databricks.
+    It uses the Databricks API to get the latest repair ID before sending the repair query.
+
+    Note that we use the `JobsService` class instead of the `RunsApi` class. This is because the
+    `RunsApi` class does not allow sending the `include_history` parameter which is necessary for
+    repair jobs.
+
+    Also for the moment we don't allow custom retry_callbacks. We might implement this in
+    the future if users ask for it, but for the moment we want to keep things simple while the API
+    stabilizes.
+    :return: None
+    """
+    current_task = context["ti"].task
+    api_client = current_task._get_api_client()
+    jobs_service = JobsService(api_client)
+    current_job = jobs_service.get_run(
+        run_id=current_task.databricks_run_id, include_history=True
+    )
+    repair_history = current_job.get("repair_history")
+    repair_history_id = None
+    if repair_history and len(repair_history) > 1:
+        # We use the last item in the array to get the latest repair ID
+        repair_history_id = repair_history[-1]["id"]
```

Comment:
We're not setting the `repair_history_id` when `len(repair_history)==1`. We would need to handle that case too.